libs/components/forms/src/lib/modules/toggle-switch/toggle-switch.component.ts

```diff
@@ -56,74 +56,84 @@ export class SkyToggleSwitchComponent
    * If the `sky-toggle-switch-label` component displays a visible label, do not use this property.
    */
   @Input()
-  public ariaLabel: string;
+  public ariaLabel: string | undefined;
 
   /**
    * Indicates whether the toggle switch is selected.
    * @default false
    */
   @Input()
-  public set checked(checked: boolean) {
-    if (checked !== this.checked) {
-      this._checked = checked;
-      this.onChange(checked);
+  public set checked(checked: boolean | undefined) {
+    if (!!checked !== this.checked) {
+      this.#_checked = !!checked;
+      this.#onChange(checked);
 
       // Do not mark the field as "dirty"
       // if the field has been initialized with a value.
-      if (this.isFirstChange && this.control) {
-        this.control.markAsPristine();
-        this.isFirstChange = false;
+      if (this.#isFirstChange && this.#control) {
+        this.#control.markAsPristine();
+        this.#isFirstChange = false;
       }
     }
   }
 
   public get checked(): boolean {
-    return this._checked || false;
+    return this.#_checked;
   }
 
   /**
    * Indicates whether to disable the toggle switch.
    */
   @Input()
-  public disabled = false;
+  public disabled: boolean | undefined = false;
 
   /**
    * Specifies a tab index for the toggle switch. If not defined, the index is set to the position
    * of the toggle switch on load.
    */
   @Input()
-  public tabIndex = 0;
+  public tabIndex: number | undefined = 0;
 
   /**
    * Fires when the checked state of a toggle switch changes.
    */
   @Output()
   public toggleChange = new EventEmitter<SkyToggleSwitchChange>();
 
-  public get hasLabelComponent(): boolean {
-    return this.labelComponents.length > 0;
-  }
+  public hasLabelComponent = false;
 
   public enableIndicatorAnimation = false;
 
   @ContentChildren(SkyToggleSwitchLabelComponent)
-  private labelComponents: QueryList<SkyToggleSwitchLabelComponent>;
+  public labelComponents: QueryList<SkyToggleSwitchLabelComponent> | undefined;
+
+  #control: AbstractControl | undefined;
+  #isFirstChange = true;
+  #ngUnsubscribe = new Subject<void>();
 
-  private control: AbstractControl;
-  private isFirstChange = true;
-  private ngUnsubscribe = new Subject<void>();
+  #_checked = false;
 
-  private _checked = false;
+  #changeDetector: ChangeDetectorRef;
 
-  constructor(private changeDetector: ChangeDetectorRef) {}
+  constructor(changeDetector: ChangeDetectorRef) {
+    this.#changeDetector = changeDetector;
+  }
 
   public ngAfterContentInit(): void {
+    // istanbul ignore if
+    if (!this.labelComponents) {
```

Comment:
I'd prefer to put the logic inside `if (this.labelComponents)` instead of throwing an error if it's not found, then using `/* istanbul ignore else */`.